packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -850,22 +873,37 @@ Future<void> _writeMacOSPluginRegistrant(FlutterProject project, List<Plugin> pl
   );
 }
 
-/// Filters out Dart-only plugins, which shouldn't be added to the native generated registrants.
-List<Plugin> _filterNativePlugins(List<Plugin> plugins, String platformKey) {
+/// Filters out Dart-only and FFI plugins, which shouldn't be added to the
+/// native generated registrants.
+List<Plugin> _filterMethodChannelPlugins(
```

Comment:
what about plugins that use both dart:ffi AND message channels with native code?